### DISCUSSION TITLE
Strange behavior when using Option73

### DISCUSSION THREAD

**Discussion by avishorp** [arendst/Tasmota#15147] [Q&A] [answered]
I'm using a SONOFF T0 Touch module with the latest Tasmota version. After setting Option73 to "ON" I get strange behavior. When the relay is in "OFF" state, pressing the corresponding button switches it to "ON". Afterwards, touching the button only sends MQTT messages without changing the relay state, as expected. If I force the relay back to "OFF", I get the same behavior. Any ideas what is wrong?

**Comment by ascillato2** [comment, marked as answer]
Set Weblog to 4 and check the console logs.

**Reply by avishorp** [reply]
What should I look for?

**Reply by barbudor** [reply]
What happens when you press the button that could lead to a clue why power is switched ON.

**Reply by avishorp** [reply]
This is the log content when the relay is OFF (the log between 17.257-17.494 is turning off the relay via the Web GUI):
```
00:06:17.257 MQT: stat/test-sw/POWER2 = OFF
00:06:17.494 CFG: Saved to flash at FB, Count 239, Bytes 4096
00:06:18.773 HTP: Console
00:06:22.685 MQT: cmnd/test-sw/POWER2 = ON
00:06:22.690 MQT: stat/test-sw/RESULT = {"Button2":{"Action":"SINGLE"}}
00:06:22.730 SRC: MQTT
00:06:22.732 CMD: Grp 0, Cmnd 'POWER', Idx 2, Len 2, Data 'ON'
00:06:22.741 MQT: tele/test-sw/STATE = {"Time":"1970-01-01T00:06:22","Uptime":"0T00:06:23","UptimeSec":383,"Heap":28,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":21,"MqttCount":1,"POWER1":"OFF","POWER2":"ON","POWER3":"OFF","Wifi":{"AP":1,"SSId":"XXXXX","BSSId":"XXXXXXX","Channel":1,"Mode":"11n","RSSI":96,"Signal":-52,"LinkCount":1,"Downtime":"0T00:00:03"}}
00:06:22.767 MQT: stat/test-sw/RESULT = {"POWER2":"ON"}
00:06:22.771 MQT: stat/test-sw/POWER2 = ON
00:06:23.338 CFG: Saved to flash at FA, Count 240, Bytes 4096
```

This is the log when the relay is already turned on:
```
00:03:59.818 MQT: cmnd/test-sw/POWER2 = ON
00:03:59.824 MQT: stat/test-sw/RESULT = {"Button2":{"Action":"SINGLE"}}
00:03:59.863 SRC: MQTT
00:03:59.866 CMD: Grp 0, Cmnd 'POWER', Idx 2, Len 2, Data 'ON'
00:03:59.875 MQT: tele/test-sw/STATE = {"Time":"1970-01-01T00:03:59","Uptime":"0T00:04:00","UptimeSec":240,"Heap":28,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":21,"MqttCount":1,"POWER1":"OFF","POWER2":"ON","POWER3":"OFF","Wifi":{"AP":1,"SSId":"XXXXXX","BSSId":"XXXXXXX","Channel":1,"Mode":"11n","RSSI":94,"Signal":-53,"LinkCount":1,"Downtime":"0T00:00:03"}}
00:03:59.903 MQT: stat/test-sw/RESULT = {"POWER2":"ON"}
00:03:59.906 MQT: stat/test-sw/POWER2 = ON
```

**Reply by ascillato** [reply]
From your log: `SRC: MQTT` means that your MQTT broker is sending the command POWER to your device. So, Setoption73 is working fine and as expected. Your issue comes from your broker.

**Reply by barbudor** [reply]
It may also be a ButtonTopic issue
Did you played with ButtonTopic before plating with SetOption73 ? WHat is your current `ButtonTopic` value ?

**Reply by avishorp** [reply]
@ascillato is right, it was all a Home Assistant mis-configuration. Thanks!
